extension/src/service/klarna-service.js

```diff
@@ -0,0 +1,86 @@
+const {
+  ADYEN_PERCENTAGE_MINOR_UNIT,
+  DEFAULT_PAYMENT_LANGUAGE,
+  KLARNA_DEFAULT_LINE_ITEM_NAME
+} = require('../config/klarna-constants')
+
+function createLineItems (payment, cart) {
+  const lineItems = []
+  const locales = _getLocales(cart, payment)
+
+  cart.lineItems.forEach((item) => {
+    lineItems.push(_createAdyenLineItemFromLineItem(item, locales))
+  })
+
+  cart.customLineItems.forEach((item) => {
+    lineItems.push(_createAdyenLineItemFromCustomLineItem(item, locales))
+  })
+
+  lineItems.push(_createShippingInfoAdyenLineItem(cart.shippingInfo))
+
+  return lineItems
+}
+
+function _getLocales (cart, payment) {
+  const locales = []
+  let paymentLanguage = payment.custom && payment.custom.fields['languageCode']
+  if (!paymentLanguage)
```

Comment:
refactoring point: these checks are not easy to reason, what happens to if-else cases or switch cases?

Reply:
I also tried to refactor this method, but I don't know how. If you have a suggestion, please write it here :)

Reply:
I don't want to spend time on this, but the review is obvious I could not understand that code, maybe sth like below (I did not check the tests)

```
if (paymentLanguage)
    locales.push(paymentLanguage)
  else
    paymentLanguage = cart.locale
    if (paymentLanguage !== DEFAULT_PAYMENT_LANGUAGE)
      locales.push(paymentLanguage)
```

Reply:
Yes, it's not easy to understand. This fallback algorithm is complex in [payone](https://github.com/commercetools/commercetools-payone-integration/blob/dfb14f4f17c356fa0d14b7f007aebe76065942ec/service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/common/AuthorizationRequestWithCart.java#L183) as well. Here I already tried to simplify it. Your suggestion is changing the method behaviour. I added a comment to make it more clear: https://github.com/commercetools/commercetools-adyen-integration/pull/294/commits/419c4a50172ef8f2d667dc57bc5fd28eaadf0703#diff-4d2f466e11952575f6b8048a0f137b16R46-R50